models/banner.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+
+const MODEL = {
+    id: Joi
+        .number()
+        .integer()
+        .positive(),
+
+    message: Joi
+        .string()
+        .max(512)
+        .description('Body of banner to display')
+        .example('Due to planned upgrade of Kubernetes, Screwdriver will be down'),
+
+    isActive: Joi
+        .boolean()
+        .description('Flag if the banner is active')
+        .example(true)
+        .default(false),
+
+    dateCreated: Joi
+        .string()
+        .isoDate()
+        .description('When this banner was created')
+        .example('2017-01-06T01:49:50.384359267Z'),
+
+    createdBy: Joi
+        .number()
+        .integer()
+        .positive()
+        .description('ID of user creating the banner'),
+
+    type: Joi
+        .string().valid([
+            'info',
+            'warn'
+        ])
+        .description('Type/Severity of the banner message')
+        .example('info')
+};
+
+module.exports = {
+    /**
+     * All the available properties of Banners
+     *
+     * @property base
+     * @type {Joi}
+     */
+    base: Joi.object(MODEL).label('Banner'),
+
+    /**
+     * Properties for Banner that will come back during a GET request
+     *
+     * @property get
+     * @type {Joi}
+     */
+    get: Joi.object(mutate(MODEL, [
+        'id'
+    ], [])).label('Get Banner'),
+
+    /**
+     * Properties for Banners that will be passed during a CREATE request
+     *
+     * @property create
+     * @type {Joi}
+     */
+    create: Joi.object(mutate(MODEL, [
+        'message'
+    ], [
+        'type'
+    ])).label('Create Banner'),
+
+    /**
+     * Properties for Banners that will be passed during a UPDATE request
+     *
+     * @property update
+     * @type {Joi}
+     */
+    update: Joi.object(mutate(MODEL, [], [
+        'id',
+        'message',
+        'type',
+        'isActive'
+    ])).label('Update Banner'),
+
+    /**
+     * Properties for Banners that will come back during a LIST request.
+     * The LIST request will list all banners
+     */
+    list: Joi.array().items(Joi.object(mutate(MODEL, [
+    ], [
```

Comment:
should probably at least return `id`
should be similar to https://github.com/screwdriver-cd/data-schema/blob/9f9d6fbf23716b278e7b0ad7ed3095973c7b2bda/models/collection.js#L76-L83

Reply:
Can also remove this section in here and just do something like https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/templates/list.js#L6 in the API repo later